flake.nix

```diff
@@ -53,6 +53,8 @@
             ];
 
             nativeBuildInputs = [ pkgs.installShellFiles ];
+
+            meta.mainProgram = "typst";
```

Comment:
Does it still work if you move it to the `typst = craneLib.buildPackage (...)` call below? I believe that could be more appropriate (right now this option would propagate to the dependency derivation as well).

Reply:
It is already in there since we're concatenating `commonCraneArgs`, so yeah!

Reply:
> It is already in there since we're concatenating `commonCraneArgs`, so yeah!

Of course; I meant that it could be more appropriate to move that line away from `commonCraneArgs` and into the `typst` derivation's options directly, because `commonCraneArgs` is also used for the `cargoArtifacts` derivation, which I believe doesn't need `meta.mainProgram` (as it does not even build it, iirc). :slightly_smiling_face:

Reply:
Right, just did it !

Reply:
Thank you :handshake: :rocket: